project/src/main/webapp/script.js

```diff
@@ -395,3 +402,29 @@ function registerUserByToken() {
   }
   fetch('/register?idToken=' + googleUser.getAuthResponse().id_token, {method: 'POST'});
 }
+
+/** Sends feedback that includes the place the user chose. */
+function sendUserChoiceAsFeedback() {
+  let indexOfPlaceUserChose = getCheckedValueByElementId('chosen-place-form',
+      'Please select the place that you chose.');
+  if (indexOfPlaceUserChose <= recommendedPlaces.length) {
+    updateUserFeedback(recommendedPlaces, recommendedPlaces[indexOfPlaceUserChose].placeId,
+        false /** user tried again */)
+  }
+}
+
+/** Updates user feedback in the database, by using the feedback servlet. */
+function updateUserFeedback(recommendedPlaces, chosenPlaceId, tryAgain) {
+  if (!googleUser) {
```

Comment:
I got this tip from Ariel on my PR and agreed with it:
write:

if (googleUser){
  logic
}

And then no "return" is needed.

Reply:
That's weird, I think a while ago I got the opposite tip from her, and the explanation was that we don't want to put the big chunk of code inside an "if". why do you think this is better?

Reply:
I think it's more elegant without the "return". But there is no right answer..

Reply:
My 2 cents: I actually think differently than the first comment suggested. 
This code is nested:
```
if (condition) {
   // Do something very long which might have other conditions 
  // and for loops etc.
}
```

But this code is less nested:
```
if (!condition) {
  // nothing to do
  return;
}
// The rest of the code is at the upper level, causing 
// less confusion for the reader who needs to know at 
// what level of nesting the code currently is
```

This is actually one of the C++ best practices at Google.